src/components/floating-emojis/FloatingEmoji.js

```diff
@@ -24,73 +26,82 @@ const FloatingEmoji = ({
-      inputRange: [0, 1],
-      outputRange: [0, distance],
+  const animation = useSharedValue(0);
+
+  useLayoutEffect(() => {
+    animation.value = withTiming(1, {
+      duration,
+      easing: Easing.linear,
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const progress = interpolate(animation.value, [0, 1], [0, distance]);
+
+    const opacity = interpolate(
+      progress,
+      [0, distance * (opacityThreshold ?? 0.5), distance - size],
+      [1, fadeOut ? 0.89 : 1, fadeOut ? 0 : 1]
+    );
+
+    const rotate =
+      interpolate(
+        progress,
+        [0, distance / 4, distance / 3, distance / 2, distance],
+        [0, -2, 0, 2, 0]
+      ) + 'deg';
+
+    const scale = interpolate(
+      progress,
+      [0, 15, 30, 50, distance],
+      [0, 1.2, 1.1, 1, scaleTo]
+    );
+
+    const everyThirdEmojiMultiplier = index % 3 === 0 ? 3 : 2;
+    const everySecondEmojiMultiplier = index % 2 === 0 ? -1 : 1;
+    const translateXComponentA =
+      animation.value *
+      size *
+      everySecondEmojiMultiplier *
+      everyThirdEmojiMultiplier;
+
+    const wiggleMultiplierA = Math.sin(progress * (distance / (350 / 15)));
+    const wiggleMultiplierB = interpolate(
+      progress,
+      [0, distance / 10, distance],
+      [10 * wiggleFactor, 6.9 * wiggleFactor, 4.2069 * wiggleFactor]
+    );
+    const translateXComponentB = wiggleMultiplierA * wiggleMultiplierB;
+
+    const translateX = disableHorizontalMovement
+      ? 0
+      : translateXComponentA + translateXComponentB;
+
+    const translateY = disableVerticalMovement ? 0 : -progress;
 
     return {
-      opacity: interpolate(progress, {
-        inputRange: [0, distance * opacityThreshold || 0.5, distance - size],
-        outputRange: [1, fadeOut ? 0.89 : 1, fadeOut ? 0 : 1],
-      }),
-      rotate: concat(
-        interpolate(progress, {
-          inputRange: [0, distance / 4, distance / 3, distance / 2, distance],
-          outputRange: [0, -2, 0, 2, 0],
-        }),
-        'deg'
-      ),
-      scale: interpolate(progress, {
-        inputRange: [0, 15, 30, 50, distance],
-        outputRange: [0, 1.2, 1.1, 1, scaleTo],
-      }),
-      translateX: add(
-        multiply(
-          animation,
-          size,
-          index % 3 === 0 ? 3 : 2,
-          index % 2 === 0 ? -1 : 1
-        ),
-        multiply(
-          sin(multiply(progress, distance / (350 / 15))), // i rly dont understand math plz help
-          interpolate(progress, {
-            inputRange: [0, distance / 10, distance],
-            outputRange: [
-              10 * wiggleFactor,
-              6.9 * wiggleFactor,
-              4.2069 * wiggleFactor,
-            ],
-          })
-        )
-      ),
-      translateY: multiply(animation, distance, -1),
+      opacity,
+      transform: [{ rotate }, { scale }, { translateX }, { translateY }],
     };
-  }, []);
+  });
```

Comment:
I think you can add an array here. AFAIK this will have some positive performance impact